File: easycontrol/server/src/main/java/top/saymzx/easycontrol/server/entity/DisplayInfo.java
/*
 * 本项目大量借鉴学习了开源投屏软件：Scrcpy，在此对该项目表示感谢
 */
package top.saymzx.easycontrol.server.entity;

import android.util.Pair;

public final class DisplayInfo {
  public final int displayId;
  public final int width;
  public final int height;
  public final int rotation;
  public final int layerStack;
  public final int density;

  public DisplayInfo(int displayId, int width,int height, int rotation,int density, int layerStack) {
    this.displayId = displayId;
    this.width=width;
    this.height=height;
    this.rotation = rotation;
    this.layerStack = layerStack;
    this.density = density;
  }
}

File: scrcpy_android/server/src/main/java/top/saymzx/scrcpy/server/entity/DisplayInfo.kt
package top.saymzx.scrcpy.server.entity

class DisplayInfo(
  val displayId: Int,
  val size: Pair<Int, Int>,
  val rotation: Int,
  val layerStack: Int,
  val flags: Int
) {

  companion object {
    const val FLAG_SUPPORTS_PROTECTED_BUFFERS = 0x00000001
  }
}